analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -36,6 +38,33 @@ public override bool ShouldExecute()
         return finder.HasPotentialNullableValueAccess;
     }
 
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var operationInstance = context.Operation.Instance;
+        if (operationInstance.Kind == OperationKindEx.PropertyReference
+            && operationInstance.ToPropertyReference() is { Instance: var instance } propertyReference
+            && propertyReference.Property.Name == nameof(Nullable<int>.Value)
+            && IsNull(instance))
+        {
+            ReportIssue(instance, instance.Syntax.ToString());
+        }
+        else if (operationInstance.Kind == OperationKindEx.Conversion
+            && operationInstance.ToConversion() is { Operand: var operand } conversion
+            && operand.Type.IsNullableValueType()
+            && !conversion.Type.IsNullableValueType()
+            && conversion.Type.IsStruct()
+            && IsNull(operand))
+        {
+            ReportIssue(operand, operand.Syntax.ToString());
+        }
+
+        return context.State;
+
+        bool IsNull(IOperation operation) =>
+            context.HasConstraint(operation, ObjectConstraint.Null)
+            && SemanticModel.GetTypeInfo(operation.Syntax).Nullability().FlowState != NullableFlowState.NotNull;
```

Comment:
What is this `FlowState` for? When I delete the line, no UT breaks. 

We either need to remove it, or add UT that will prove it has a value we can't detect.

Reply:
I will reintroduce when and if I find a valid scenario for it.